cg/meta/upload/gt.py

```diff
@@ -43,93 +47,168 @@ def data(self, analysis: Analysis) -> dict:
+        samples_sex: dict[str, dict[str, str]] = {}
+        for sample in case.samples:
+            sample_id: str = sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": sample.sex,
+                "analysis": UploadGenotypesAPI._get_analysis_sex_raredisease(
+                    qc_metrics_file=qc_metrics_file, sample_id=sample_id
+                ),
+            }
+        return samples_sex
 
     @staticmethod
-    def _is_variant_file(genotype_file: File):
-        return genotype_file.full_path.endswith("vcf.gz") or genotype_file.full_path.endswith("bcf")
+    def _get_analysis_sex_raredisease(qc_metrics_file: Path, sample_id: str) -> str:
+        """Return analysis sex for each sample of an analysis."""
+        qc_metrics: list[MetricsBase] = (
+            UploadGenotypesAPI._get_parsed_qc_metrics_deliverables_raredisease(qc_metrics_file)
+        )
+        for metric in qc_metrics:
+            if metric.name == RAREDISEASE_PREDICTED_SEX_METRIC and metric.id == sample_id:
+                return str(metric.value)
 
-    def _get_genotype_files(self, version_id: int) -> list:
-        return self.hk.files(version=version_id, tags=["genotype"]).all()
+    @staticmethod
+    def _get_parsed_qc_metrics_deliverables_raredisease(qc_metrics: Path) -> list[MetricsBase]:
+        """Parse and return a QC metrics file."""
+        qcmetrics_raw: dict = ReadFile.get_content_from_file(
+            file_format=FileFormat.YAML, file_path=qc_metrics
+        )
+        return [MetricsBase(**metric) for metric in qcmetrics_raw["metrics"]]
 
     @staticmethod
-    def is_suitable_for_genotype_upload(case_obj: Case) -> bool:
-        """Check if a cancer case is contains WGS and normal sample."""
+    def _is_variant_file(genotype_file: File):
+        return genotype_file.full_path.endswith(
+            FileExtensions.VCF_GZ
+        ) or genotype_file.full_path.endswith(FileExtensions.BCF)
 
-        samples: list[Sample] = case_obj.samples
-        return any(
-            (not sample.is_tumour and PrepCategory.WHOLE_GENOME_SEQUENCING == sample.prep_category)
-            for sample in samples
-        )
+    @staticmethod
+    def _sort_genotype_files(hk_genotype_files: list[File]) -> File | None:
+        """
+        Take a list of files and only keep files finishing with .bcf or .vcf.gz
+        Returns a single remaining file or raises ValueError if more than one file remains.
+        """
+        allowed_extensions = (FileExtensions.BCF, FileExtensions.VCF_GZ)
+        filtered_files = [
+            file for file in hk_genotype_files if file.full_path.endswith(allowed_extensions)
+        ]
+        if len(filtered_files) > 1:
+            raise ValueError(
+                f"Error: Expected only one genotype file, but found {len(filtered_files)} "
+                f"({', '.join(map(str, filtered_files))})."
+            )
+        return filtered_files[0] if filtered_files else None
```

Comment:
```suggestion
    def _get_single_genotype_file(hk_genotype_files: list[File]) -> File:
        """
        Returns the single .bcf or .vcf.gz file expected to be found in the provided list. Raises an error if the amount of such files is different from 1.
        """
        allowed_extensions = (FileExtensions.BCF, FileExtensions.VCF_GZ)
        filtered_files = [
            file for file in hk_genotype_files if file.full_path.endswith(allowed_extensions)
        ]
        if len(filtered_files) != 1:
            raise ValueError(
                f"Error: Expecte one genotype file, but found {len(filtered_files)} "
                f"({', '.join(map(str, filtered_files))})."
            )
        return filtered_files[0]
```

Reply:
Mixed return types are often difficult to work with and it seemed as if there has been an error if no file is found either way.